modules/core/src/utils/prop-equal.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Fast partial deep equal for prop.
+ *
+ * @param a Prop
+ * @param b Prop to compare against `a`
+ * @param depth Depth to which to recurse in nested Objects/Arrays. Use 0 (default) for shallow comparison, -1 for infinite depth
+ */
+/* eslint-disable complexity */
+export function propEqual(a: any, b: any, depth: number): boolean {
+  if (a === b) {
+    return true;
+  }
+  if (
+    !a ||
+    !b ||
+    !(a.constructor === Array || a.constructor === Object) ||
+    !(b.constructor === Array || b.constructor === Object)
```

Comment:
```suggestion
    Array.isArray(a) !== Array.isArray(b) ||
    typeof a !== 'object' ||
    typeof b !== 'object'
```

A string will pass through to the for loop unless you check the argument types.